packages/components/phone-number/src/components/osds-phone-number/osds-phone-number.tsx

```diff
@@ -51,35 +61,81 @@ export class OsdsPhoneNumber implements OdsPhoneNumberAttribute {
     this.isoCode = this.controller.getDefaultIsoCode();
     this.locale = this.controller.getDefaultLocale();
     this.handlerLocale(this.locale);
+    this.isValidValue(this.value);
   }
 
   @Watch('locale')
   handlerLocale(locale: ODS_LOCALE): void {
     const translationFile = this.controller.loadTranslationFileByLocale(locale);
     this.i18nCountriesMap = translationFile.reduce((acc, country) => {
-      acc.set(country.isoCode, country);
+      const exampleNumber = this.phoneUtil.getExampleNumber(country.isoCode);
+      acc.set(country.isoCode, { ...country, countryCode: exampleNumber?.getCountryCode() });
       return acc;
     }, new Map());
   }
 
   @Watch('countries')
   handlerCountries(): void {
     const countriesList = this.controller.getCountriesList();
-    this.parseCountries(countriesList);
+    this.parsedCountries = this.controller.parseCountries(countriesList);
     this.hasCountries = !!this.parsedCountries?.length;
   }
 
-  private parseCountries(countries: readonly ODS_COUNTRY_ISO_CODE[] | string) {
-    if (typeof countries === 'string') {
-      try {
-        this.parsedCountries = JSON.parse(countries);
-      } catch (err) {
-        this.logger.warn('[OsdsPhoneNumber] countries string could not be parsed.');
-        this.parsedCountries = [];
-      }
-    } else {
-      this.parsedCountries = [...countries];
+  @Listen('odsValueChange')
+  handlerOdsValueChange(event: CustomEvent<OdsInputValueChangeEventDetail | OdsInputPhoneNumberValueChangeEventDetail | OdsSelectValueChangeEventDetail>): void {
+    if ('isoCode' in event.detail) {
+      return;
+    }
+    event.preventDefault();
+    event.stopPropagation();
+    if ('selection' in event.detail) {
+      this.isoCode = ODS_COUNTRY_ISO_CODES.find((countryIsoCode) => countryIsoCode === event.detail.value);
+      this.value = '';
+      this.error = false;
+      return;
+    }
+    return this.handlerInputEvent(event.detail);
+  }
+
+  private handlerInputEvent(event: OdsInputValueChangeEventDetail) {
+    if (event.value) {
+      this.value = event.value;
+    }
+    if(!this.isValidValue(this.value)) {
+      return;
+    }
+    const number = this.controller.parseNumber(event.value);
+    const oldNumber =this.controller.parseNumber(event.oldValue);
+    this.odsValueChange.emit({
+      ...event,
+      value: number && this.phoneUtil.format(number, PhoneNumberFormat.E164),
+      oldValue: oldNumber && this.phoneUtil.format(oldNumber, PhoneNumberFormat.E164) || undefined,
+      isoCode: this.isoCode,
+    })
+  }
+
+  @Watch('value')
+  private isValidValue(value: string | null): boolean {
```

Comment:
would be nice to split this between:
- the validity check, who return a boolean
- the error attribute update

otherwise code like that starts to be confusing:
```
if (this.isValidValue(this.value)) {
  // here we don't expect to have updated any component attribute
}
```